storm.yaml

```diff
@@ -4,27 +4,36 @@
 mesos.master.url: "zk://localhost:2181/mesos"
 storm.zookeeper.servers:
     - "localhost"
-nimbus.host: "localhost"
 # -----------------------------------------------------------
 
 
-### If this is not false then you need to add about 128m of mem usage to executor to account for the logviewer
-#supervisor.autostart.logviewer: false
-topology.mesos.executor.mem.mb: 1128  ## default is 1000
+# Worker resources
+topology.mesos.worker.cpu: 1.0
+### Worker heap with 20% overhead
+topology.mesos.executor.mem.mb: 1250
+worker.childopts: "-Xmx1000m"
 
-# The default behavior is to launch the logvierwer unless autostart is false
-logviewer.port: 8000
+# Supervisor resources
+topology.mesos.executor.cpu: 0.1
+topology.mesos.executor.mem.mb: 500 # Supervisor memory, with 20% overhead
+supervisor.childopts: "-Xmx256m"
+
+# The default behavior is to launch the logviewer unless autostart is false.
+# If you enable the logviewer, you'll need to add memory overhead to the
+# executor for the logviewer.
 logviewer.childopts: "-Xmx128m"
 logviewer.cleanup.age.mins: 10080
 logviewer.appender.name: "A1"
-
-
-# You should not need to change anything below this line
-#--------------------------------------------------------
+supervisor.autostart.logviewer: true
```

Comment:
I spent some time this morning going through the Storm source to see how much effort it'd be to add the logviewer port to the supervisor heartbeats.

Suffice to say, all the parentheses made it a little difficult to figure out what's what. I might take another stab at it later.

Since it'd require patching Storm, I think I'll make it so the logviewer will launch only if the port is available, and otherwise not. This should let you run multiple supervisors, but not from the same Storm cluster.